### PR TITLE
Support the new compression argument in netCDF4 > 1.6.0

### LINKED ISSUE #6929: Support new netcdf4 1.6.0 compression arguments
### Is your feature request related to a problem?

When using the netcdf4 engine, I am not able to use the new "compression" argument to choose a compression scheme different from zlib in the encoding.

```
        if raise_on_invalid:
            invalid = [k for k in encoding if k not in valid_encodings]
            if invalid:
>               raise ValueError(
                    f"unexpected encoding parameters for {backend!r} backend: {invalid!r}. Valid "
                    f"encodings are: {valid_encodings!r}"
                )
E               ValueError: unexpected encoding parameters for 'netCDF4' backend: ['compression']. Valid encodings are: {'fletcher32', 'zlib', 'contiguous', 'dtype', 'least_significant_digit', 'shuffle', 'complevel', '_FillValue', 'chunksizes'}

../../../../netCDF4_.py:279: ValueError
```

Furthermore, according to the [release notes of 1.6.0](https://github.com/Unidata/netcdf4-python/blob/master/Changelog), zlib argument is to be deprecated:

```
* add 'compression' kwarg to createVariable to enable new compression
   functionality in netcdf-c 4.9.0.  'None','zlib','szip','zstd','bzip2'
   'blosc_lz','blosc_lz4','blosc_lz4hc','blosc_zlib' and 'blosc_zstd'
   are currently supported. 'blosc_shuffle',
   'szip_mask' and 'szip_pixels_per_block' kwargs also added.
   compression='zlib' is equivalent to (the now deprecated) zlib=True.
   If the environment variable NETCDF_PLUGIN_DIR is set to point to the
   directory with the compression plugin lib__nc* files, then the compression plugins will
   be installed within the package and be automatically available (the binary
   wheels have this).  Otherwise, the environment variable HDF5_PLUGIN_PATH
   needs to be set at runtime  to point to plugins in order to use the new compression
   options.
```

I am using the last versions

```
xarray: 2022.6.0
pandas: 1.4.2
numpy: 1.22.4
scipy: None
netCDF4: 1.6.0
pydap: None
h5netcdf: None
h5py: None
Nio: None
zarr: None
cftime: 1.6.0
nc_time_axis: None
PseudoNetCDF: None
rasterio: None
cfgrib: None
iris: None
bottleneck: None
dask: 2022.6.0
distributed: 2022.6.0
matplotlib: 3.5.2
cartopy: None
seaborn: None
numbagg: None
fsspec: 2022.5.0
cupy: None
pint: None
sparse: None
flox: None
numpy_groupies: None
setuptools: 62.6.0
pip: 22.1.2
conda: None
pytest: 7.1.2
IPython: 8.4.0
sphinx: 5.1.1
```

### Describe the solution you'd like

Update the netcdf4 backend to support these arguments. Should not be too difficult.

### Describe alternatives you've considered

_No response_

### Additional context

I can try to do this myself, it does not look hard.

### PR DESCRIPTION
**Pull request by markelg**
- [x] Closes #6929
- [x] Tests added
- [ ] User visible changes (including notable bug fixes) are documented in `whats-new.rst`

### CODE REVIEW

**Comment by markelg** [comment]
I think this is ready, unfortunately it does not make sense to merge it yet, and one test is failing, because netcdf-c 4.9.0 is not yet available in conda-forge (see https://github.com/conda-forge/libnetcdf-feedstock/pull/140). So I would put it on hold.

File: xarray/backends/netCDF4_.py [hunk 1]
```diff
@@ -242,13 +243,21 @@ def _extract_nc4_variable_encoding(
...
-        valid_encodings.add("compression")
         valid_encodings.add("compression_opts")
 
+    if backend == "netCDF4" and has_netcdf4:
+        # if using netCDF4 check the versions are updated enough for supporting the new
+        # compression
+        if Version(netCDF4.__version__) < Version("1.6.0") or Version(
+            netCDF4.getlibversion().split(" ")[0]
+        ) < Version("4.9.0"):
+            valid_encodings.remove("compression")
```
Comment (dcherian) [inline xarray/backends/netCDF4_.py:259]: IMO we can just let netCDF raise the error for this case. 

We should pass along `compression` if it exists but not add or remove it by default.

**Comment by headtr1ck** [comment]
> I think this is ready, unfortunately it does not make sense to merge it yet, and one test is failing, because netcdf-c 4.9.0 is not yet available in conda-forge (see https://github.com/conda-forge/libnetcdf-feedstock/pull/140). So I would put it on hold.

Should exist now.

**Comment by dcherian** [comment]
Thanks @markelg!

**Comment by markelg** [comment]
Sorry for the delay, I completed all the remaining tasks. I hope is ready now. There are some error in the tests but look unrelated to this. There seems to be an assert equal failing because of a nan and nan !=nan.

**Pull request merged**